app/main.c

```diff
@@ -171,7 +171,7 @@ static void run_at(uintptr_t addr, int my_cpu)
     __asm__ __volatile__("movl %0, %%edi" : : "r" (new_start_addr));
 #endif
 
-    goto *new_start_addr;
+    ((void (*)(void))new_start_addr)();
```

Comment:
This makes the intent less clear. Maybe add a comment?

Reply:
Alright, done.